include/bluetooth/mesh/model_types.h

```diff
@@ -36,6 +36,7 @@ extern "C" {
 struct bt_mesh_model_transition {
 	uint32_t time; /**< Transition time value in milliseconds */
 	uint32_t delay; /**< Message execution delay in milliseconds */
+	bool from_msg; /**< Transition time from message. Set by the stack. */
```

Comment:
I prefer the `NULL` version, over this, to be honest. It parses as a proper English sentence in the set-messages (e.g. `if (set->transition) {  gradual_change(); }`), and didn't introduce any confusing parameters in the client model interfaces. The `NULL` check requirement was handled nicely by your `transition_remaining()` function, too, so as long as we use that in the samples, I don't think it's problematic. Is there some usage scenario that can't work with the NULL pointer approach?

Reply:
This fixes the issue that @alxelax has pointed out. Nothing prevents a user to instantiate Generic DTT server on Light LC element (through Scene Server for example). Then, the previous approach won't work, because `bt_mesh_dtt_srv_transition_get` will set transition time.

Reply:
I unresolved that conversation and added a comment to it. I don't think the issue Alex mentions is valid, but I can't see what you changed there anymore.

Reply:
Squashing of commits does not keep the history of changes. There are no changes anymore that have been commented on. Nevertheless, I still guess the issue that I emphasized was valid. It is a bad idea to make the assumption in some internal model algorithm based on the presence or absence third model. The combination of models might be arbitrary because of extensions even to not consider custom usage of generic models.

Reply:
I had a call with Pavel, and he explained the issue. I misunderstood the problem, but I get it now, and you're right. Pavel is doing some investigation to see if we can solve this without affecting the entire model API